x/ccv/provider/keeper/key_assignment_test.go

```diff
@@ -189,24 +189,50 @@ func TestGetAllValidatorsByConsumerAddr(t *testing.T) {
 
 func TestConsumerAddrsToPruneCRUD(t *testing.T) {
 	chainID := consumer
-	consumerAddr := types.NewConsumerConsAddress([]byte("consumerAddr1"))
-	vscID := uint64(1)
+	consumerAddr1 := types.NewConsumerConsAddress([]byte("consumerAddr1"))
+	consumerAddr2 := types.NewConsumerConsAddress([]byte("consumerAddr2"))
 
 	keeper, ctx, ctrl, _ := testkeeper.GetProviderKeeperAndCtx(t, testkeeper.NewInMemKeeperParams(t))
 	defer ctrl.Finish()
 
-	addrsToPrune := keeper.GetConsumerAddrsToPrune(ctx, chainID, vscID).Addresses
+	ts1 := ctx.BlockTime()
+	ts2 := ts1.Add(time.Hour)
+
+	addrsToPrune := keeper.GetConsumerAddrsToPrune(ctx, chainID, ts1).Addresses
 	require.Empty(t, addrsToPrune)
 
-	keeper.AppendConsumerAddrsToPrune(ctx, chainID, vscID, consumerAddr)
+	keeper.AppendConsumerAddrsToPrune(ctx, chainID, ts1, consumerAddr1)
+
+	addrsToPrune = keeper.GetConsumerAddrsToPrune(ctx, chainID, ts1).Addresses
+	require.NotEmpty(t, addrsToPrune, "addresses to prune is empty")
+	require.Len(t, addrsToPrune, 1, "addresses to prune is not len 1")
+	require.Equal(t, addrsToPrune[0], consumerAddr1.ToSdkConsAddr().Bytes())
+
+	keeper.AppendConsumerAddrsToPrune(ctx, chainID, ts2, consumerAddr2)
 
-	addrsToPrune = keeper.GetConsumerAddrsToPrune(ctx, chainID, vscID).Addresses
+	addrsToPrune = keeper.GetConsumerAddrsToPrune(ctx, chainID, ts2).Addresses
+	require.NotEmpty(t, addrsToPrune, "addresses to prune is empty")
+	require.Len(t, addrsToPrune, 1, "addresses to prune is not len 1")
+	require.Equal(t, addrsToPrune[0], consumerAddr2.ToSdkConsAddr().Bytes())
+
+	keeper.DeleteConsumerAddrsToPrune(ctx, chainID, ts1)
+	addrsToPrune = keeper.GetConsumerAddrsToPrune(ctx, chainID, ts1).Addresses
+	require.Empty(t, addrsToPrune, "addresses to prune was returned")
+	addrsToPrune = keeper.GetConsumerAddrsToPrune(ctx, chainID, ts2).Addresses
 	require.NotEmpty(t, addrsToPrune, "addresses to prune is empty")
 	require.Len(t, addrsToPrune, 1, "addresses to prune is not len 1")
-	require.Equal(t, addrsToPrune[0], consumerAddr.ToSdkConsAddr().Bytes())
+	require.Equal(t, addrsToPrune[0], consumerAddr2.ToSdkConsAddr().Bytes())
 
-	keeper.DeleteConsumerAddrsToPrune(ctx, chainID, vscID)
-	addrsToPrune = keeper.GetConsumerAddrsToPrune(ctx, chainID, vscID).Addresses
+	keeper.AppendConsumerAddrsToPrune(ctx, chainID, ts1, consumerAddr1)
+
+	addrsToPrune = keeper.ConsumeConsumerAddrsToPrune(ctx, chainID, ts2).Addresses
+	require.NotEmpty(t, addrsToPrune, "addresses to prune was returned")
+	require.Len(t, addrsToPrune, 2, "addresses to prune is not len 2")
```

Comment:
Could there be a unit test where *not* all the consumer addresses are consumed by ConsumeConsumerAddrsToPrune?
I'm not sure there's actually a test that checks that it respects the timestamp correctly?